### ISSUE TITLE
Boost on Cray

### ISSUE THREAD

**Issue by jgonthier** [psi4/psi4public#196]
What is the proper way to specify the compiler in Boost ? I modified the user-config.jam file to include the following lines:

using intel : linux : /opt/cray/craype/2.2.1/bin/CC ; using mpi : : <find-shared-library>mpich ;

but then, the build fails and I get the following complaint:

error: at project-config.jam:12
error: intel-linux initialization: parameter 'version' inconsistent
error: an explicit value was specified in an earlier initialization
error: no value is specified now

Apparently this is because project-config.jam contains the following:

if ! intel-linux in [ feature.values <toolset> ]
{
    using intel-linux ; # <= this is line 12
}

in my understanding, the above condition should make sure we are not trying to redefine intel-linux when it has been specified in user-config.jam, is that correct ?

@robertodr, do you know how I could solve this problem?

**Comment by robertodr** [comment]
@jgonthier use another name instead on `linux` in `user-config.jam` Like this:
```
using intel : linux-cray : /opt/cray/craype/2.2.1/bin/CC ; using mpi : : mpich ;

```
or whatever else. Be also sure that the MPI settings are passed correctly: http://www.boost.org/doc/libs/1_59_0/doc/html/mpi/getting_started.html#mpi.config
Let me know how it goes.

**Comment by jgonthier** [comment]
@robertodr I used the line you suggested. I guess that I should then specify
--toolset=intel-linux-cray ?

The bootstrap script does not like that, it issues the error:

### Unknown toolset: intel-linux-cray
###
### You can specify the toolset as the argument, i.e.:
###     ./build.sh gcc
###
### Toolsets supported by this script are:
###     acc, como, darwin, gcc, intel-darwin, intel-linux, kcc, kylix,
###     mipspro, mingw(msys), pathscale, pgi, qcc, sun, sunpro, tru64cxx, vacpp

and I believe that if I specify --toolset=intel-linux, it then just uses icc instead of the proper CC wrapper on the Cray.

**Comment by robertodr** [comment]
Boost compilation is really a royal pain... Can you try this:
```
using intel-linux : : /opt/cray/craype/2.2.1/bin/CC ; using mpi : : mpich ;
```
it is supposed to work, but I tried on my laptop with two different versions of GCC installed and it's always the default version that gets picked.
You might want to give a shot also to the suggestion in the third post in this forum thread:
http://boost.2283326.n4.nabble.com/MPI-how-to-compile-Boost-MPI-on-Cray-Linux-td4634179.html

**Comment by jgonthier** [comment]
I tried this, but then it is back to complaining about intel-linux being redefined in project-config.jam. This is really weird since project-config.jam has a conditional that is apparently supposed to avoid this, but it does not work for some reason.

So, I solved that by adding a command to boost/CMakeLists.txt that modifies project-config.jam with sed to eliminate the redefinition of intel-linux. So I am closing this issue.
